### PR TITLE
Document Phase 1 chat reliability policy

### PR DESCRIPTION
**Pull request by bcm**
## Summary
- Add ADR 0017 documenting Phase 1 web chat retry and failure semantics.
- Point architecture and SSE docs at the policy.
- Add conversation tests for retryable transport boundaries, non-retryable provider/server errors, and exhausted retry budget failure persistence.

## Validation
- `npm test -- --run test/conversation.test.ts -t "retry|provider|transient|failure"`
- `npm run check`
- `/review`: clean, no findings

Fixes SQR-86

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive web chat reliability policy documentation including retry behavior rules, error handling, and persistence semantics.
  * Updated architecture and SSE contract documentation to clarify chat turn retry constraints.

* **Tests**
  * Expanded conversation retry test coverage for transport failures, server errors, and exhausted retry scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/maz-org/squire/pull/393?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: docs/adr/0017-phase-1-web-chat-reliability-policy.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+type: ADR
+id: '0017'
+title: 'Phase 1 web chat reliability policy'
+status: active
+date: 2026-05-17
+---
+
+## Context
+
+SQR-7's engineering review left the web chat reliability policy underspecified.
```
[the remark this answers is not in the document]
Comment (bcm) [reply docs/adr/0017-phase-1-web-chat-reliability-policy.md]: Fixed in ab7ec48. I kept the SQR-7 provenance because that review originally surfaced the gap, and added SQR-86 as the current follow-through ticket for traceability.
